src/jxl.imageio/jxlinput.cpp

```diff
@@ -0,0 +1,370 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+// JPEG XL
+
+// https://jpeg.org/jpegxl/index.html
+// https://jpegxl.info
+// https://jpegxl.info/test-page
+// https://people.csail.mit.edu/ericchan/hdr/hdr-jxl.php
+// https://saklistudio.com/jxltests
+// https://thorium.rocks
+// https://bugs.chromium.org/p/chromium/issues/detail?id=1451807
+
+#include <algorithm>
+#include <cassert>
+#include <cstdio>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <jxl/decode.h>
+#include <jxl/decode_cxx.h>
+#include <jxl/resizable_parallel_runner_cxx.h>
+
+OIIO_PLUGIN_NAMESPACE_BEGIN
+
+#define DBG if (0)
+
+class JxlInput final : public ImageInput {
+public:
+    JxlInput() { init(); }
+    ~JxlInput() override { close(); }
+    const char* format_name(void) const override { return "jxl"; }
+    int supports(string_view feature) const override
+    {
+        return (feature == "exif" || feature == "ioproxy");
```

Comment:
Check the list of features in imageio.h's documentation of the `supports()` method. Are there others that you already support? Do you really already support exif now?

Reply:
I intend to add Exif support based upon my experience with Exiv2 development. It is little bit complicated, but doable sans the Brotli compression of metadata.

Reply:
OK, that's fine, but let's remove it from the reported feature list until it's done.